### PR TITLE
Add `panicking::panicking()` function

### PR DESCRIPTION
**Pull request by nick-mobilecoin**
The `panicking::panicking()` function is a crate function that indicates
if the current thread is panicking.

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #14** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/14" target="_blank">Graphite</a> 
    * **PR #13** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/13" target="_blank">Graphite</a> 
      * **PR #15** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/15" target="_blank">Graphite</a> 
        * **PR #22** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/22" target="_blank">Graphite</a> 
          * **PR #24** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/24" target="_blank">Graphite</a> 
            * **PR #30** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/30" target="_blank">Graphite</a> 
              * **PR #31** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/31" target="_blank">Graphite</a> 
                * **PR #32** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/32" target="_blank">Graphite</a> 
                  * **PR #33** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/33" target="_blank">Graphite</a>  👈
        * **PR #21** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/21" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/33?utm_source=stack-comment).

File: panic/src/panicking.rs [hunk 1]
```diff
@@ -115,3 +120,46 @@ pub(crate) mod panic_count {
         }
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    /// Sets panic count is 0
```
Comment (samdealy) [inline panic/src/panicking.rs:128]:
```suggestion
    /// Sets panic count to 0
```

File: panic/src/panicking.rs [hunk 2]
```diff
@@ -115,3 +120,46 @@ pub(crate) mod panic_count {
...
+    /// Similar to the tests in mod `panic_count` tests need to call this prior
```
Comment (samdealy) [inline panic/src/panicking.rs:129]:
```suggestion
    /// Similar to the tests in mod `panic_count.` Tests need to call this prior
```

**Review by samdealy — APPROVED** [review]
LGTM with minor nits

**Review by varsha888 — APPROVED** [review]

**Comment by nick-mobilecoin** [comment]
⏳ This pull request is set to merge as part of a Graphite merge job
Stack job ID: 174FRuRmqv3ul7Wrh6cz.
See details on [graphite.dev](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/33)

**Pull request merged**